common/persistence/sql/task.go

```diff
@@ -481,38 +481,43 @@ func (m *sqlTaskManager) UpdateTaskQueueUserData(ctx context.Context, request *p
 		return serviceerror.NewInternal(fmt.Sprintf("failed to parse namespace ID as UUID: %v", err))
 	}
 	err = m.txExecute(ctx, "UpdateTaskQueueUserData", func(tx sqlplugin.Tx) error {
-		err := tx.UpdateTaskQueueUserData(ctx, &sqlplugin.UpdateTaskQueueDataRequest{
-			NamespaceID:   namespaceID,
-			TaskQueueName: request.TaskQueue,
-			Data:          request.UserData.Data,
-			DataEncoding:  request.UserData.EncodingType.String(),
-			Version:       request.Version,
-		})
-		if m.Db.IsDupEntryError(err) {
-			return &persistence.ConditionFailedError{Msg: err.Error()}
-		}
-		if err != nil {
-			return err
-		}
-		if len(request.BuildIdsAdded) > 0 {
-			err = tx.AddToBuildIdToTaskQueueMapping(ctx, sqlplugin.AddToBuildIdToTaskQueueMapping{
+		for taskQueue, update := range request.Updates {
```

Comment:
This could be up to 100 separate Update writes, right? I assume self-hosted users won't usually see a lot of parallel updates?

Reply:
It's nothing to do with self-hosted or not, it's how many task queues are spanned by a versioning-3 deployment. Updating a deployment has to change data on some/all of its task queues at once, which are all in the same namespace, which means same cass partition, which means contention on the lwt